pkg/kotsadm/main.go

```diff
@@ -168,6 +166,40 @@ func Deploy(deployOptions types.DeployOptions) error {
 	return nil
 }
 
+func canUpgrade(upgradeOptions types.UpgradeOptions, clientset *kubernetes.Clientset, log *logger.Logger) error {
+	_, err := clientset.CoreV1().Namespaces().Get(upgradeOptions.Namespace, metav1.GetOptions{})
+	if kuberneteserrors.IsNotFound(err) {
+		err := errors.New("The namespace cannot be found or accessed")
+		return err
+	}
+
+	if upgradeOptions.ForceUpgradeKurl {
+		return nil
+	}
+
+	// don't upgrade kurl clusters.  kurl only installs into default namespace.
+	if upgradeOptions.Namespace != "" && upgradeOptions.Namespace != "default" {
```

Comment:
@divolgin can we also check the the existence of the kurl.sh node label?  Require both the node label and the default namespace?

Reply:
@marccampbell the node that the pods are on? or just all nodes?

Reply:
all nodes